zlib/http.go

```diff
@@ -76,13 +76,16 @@ type HTTPResponse struct {
 	BodySHA256   []byte      `json:"body_sha256,omitempty"`
 }
 
+type HTTP struct {
+	ProxyRequest         *HTTPRequest           `json:"connect_request,omitempty"`
+	ProxyResponse        *HTTPResponse          `json:"connect_response,omitempty"`
+	Response             *HTTPResponse          `json:"response,omitempty"`
+	RequestResponseChain []*HTTPRequestResponse `json:"request_response_chain,omitempty"`
+}
+
 type HTTPRequestResponse struct {
-	ProxyRequest      *HTTPRequest    `json:"connect_request,omitempty"`
-	ProxyResponse     *HTTPResponse   `json:"connect_response,omitempty"`
-	Request           *HTTPRequest    `json:"request,omitempty"`
-	Response          *HTTPResponse   `json:"response,omitempty"`
```

Comment:
Yes - we previously output both the initial request and the initial response (and proxy request/response, if present). The new format loses the explicit `Request` field, but the initial request can be found in the first HTTPRequestReponse element in the HTTP.RequestResponseChain. The somewhat confusing part is that I renamed the original HTTPRequestResponse struct to simply HTTP, and then I also created a new struct called HTTPRequestResponse that only contains a Request/Response pair.